docs: update conflict detection and resolution docs (#724)
aerogear/mobile-docs: modules/ROOT/pages/_partials/data-sync/conflicts-introduction.adoc

[id="conflict-resolution-introduction"]

= Introduction

> Applications that allow users to modify data while offline most likely have to deal with conflicts. A *conflict* occurs whenever two or more clients try to modify the same data in between synchronizations.

Reviewer: ```suggestion
Applications that allow users to modify data while offline must deal with conflicts. A *conflict* occurs when two or more clients try to modify the same data in between synchronizations.
```

> *Example:* A user tried to modify a record while they were offline. When they came back online, they discovered that the record was already deleted.

Reviewer: ```suggestion
*Example:* A user modifies a record while offline. When back online, they discover that the record has been deleted by another user.
```

> When talking about conflicts, it can usually be broken down into two parts.

Reviewer: ```suggestion
Discussions about conflicts can be broken down into two parts.
```

* *Conflict Detection* is the ability of an application to detect the presence of outdated data which cannot be stored without possible data loss occuring.

* *Conflict Resolution* is how the application handles the conflict and ensures the correct data is stored.

In most cases, the way conflicts are detected and resolved are specific to an application and the underlying data storage.

> In a GraphQL server, conflict detection happens exclusively in *mutations*.

Reviewer 2: Data Sync server?

Author: Perfect. Will add that.

> {sync-server} provides mechanisms for conflict detection while the {sync-service} SDK provides some utilities to help applications resolve conflicts client-side.

Reviewer 2: struggling to translate this

Author: any part in particular? 

At a high level its basically: 

Server -> detection
Client -> resolution.

If you had a suggestion on how best to word it that would be great.

Reviewer 3: Suggestion: "Data Sync server provides conflict detection on the server side while Data Sync Client provides conflict resolution on the client side."
